sgt-back-end/index.js

```diff
@@ -0,0 +1,118 @@
+const pg = require('pg');
+const express = require('express');
+const app = express();
+
+app.use(express.json());
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      res.status(200).json(result.rows);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'Query failed' });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const name = req.body.name;
+  const courseName = req.body.course;
+  const score = Number(req.body.score);
+  const params = [name, courseName, score];
+  const sql = `
+    insert into "grades" ("name", "course", "score")
+    values ($1, $2, $3)
+    returning *
+  `;
+  if (!Number.isInteger(score) || score <= 0 || score > 100) {
+    res.status(400).json({ error: 'score must be an integer between 0 and 100' });
+  }
```

Comment:
Be sure to `return` from your guards to stop the rest of the route from executing. As written, your code still attempts to query the database and respond a second time if the score is invalid, like `-10`.